pxtlib/auth.ts

```diff
@@ -347,6 +349,9 @@ namespace pxt.auth {
                 } else {
                     pxt.reportError("identity", "update preferences failed", result as any);
                 }
+                if (callback) {
+                    callback(result.success, result.resp)
```

Comment:
hmmmm my thinking was I don't know how to return a promise in the case where we're debouncing the set preferences call? but if there's a way I will switch

Reply:
Agreed, it would be better to use the returned promise. I wonder if it will work correctly for debounced saves, but could sort that out later. This is an immediate save so should be fine. Update the promise to return a success/fail boolean.

Reply:
For debouncing, keep a promise around and just return it if it already exists, otherwise create it. We do this in a few places. cloud.ts syncAsync is one example. I think it's ok to not add that today, since we don't have a test case. Just leave a comment that it needs implementing.

Reply:
I'll try it out! I do want to mention -- there is no promise returned here currently, but I'll add it.

Reply:
it's an async function so a promise is implicitly returned. You can just put a `return` here

Reply:
ooooooooooooooooo i see!